test/core/end2end/tests/retry_exceeds_buffer_size_in_subsequent_batch.cc

```diff
@@ -109,9 +109,10 @@ static void test_retry_exceeds_buffer_size_in_subsequent_batch(
   grpc_metadata_array trailing_metadata_recv;
   grpc_metadata_array request_metadata_recv;
   grpc_call_details call_details;
-  char buf[102401];
+  char* buf = static_cast<char*>(gpr_malloc(102401 * sizeof(*buf)));
   memset(buf, 'a', sizeof(buf) - 1);
   buf[sizeof(buf) - 1] = '\0';
+  // TODO(markdroth): buf is not a static string, so fix the next line
```

Comment:
Instead could you do something like:

grpc_slice request_payload_slice = grpc_slice_maalloc(102401)
memset(GRPC_SLICE_BEGIN(request_payload_slice), 'a', GRPC_SLICE_LENGTH(request_payload_slice) - 1);
*(GRPC_SLICE_BEGIN(request_payload_slice) + GRPC_SLICE_LENGTH(request_payload_slice) - 1) = '\0'

Reply:
Or, nevermind, I guess since this is a test, and the malloc-ed buf lives till the end, `grpc_slice_from_static_string` is safe enough... but is feels dirty

Reply:
It feels dirty and I hate it.